react/features/virtual-backgrounds/actions.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import { getLocalVideoTrack } from '../../features/base/tracks';
+
+import { BACKGROUND_DISABLED, BACKGROUND_ENABLED } from './actionTypes';
+import { getBackgroundEffect } from './functions';
+import logger from './logger';
+
+/**
+ * Signals the local participant activate the virtual background video or not.
+ *
+ * @param {boolean} enabled - If true enables video blur, false otherwise.
+ * @returns {Promise}
+ */
+export function toggleBackgroundEffect(enabled: boolean) {
+    return async function(dispatch: Object => Object, getState: () => any) {
+        const state = getState();
+
+        const { jitsiTrack } = getLocalVideoTrack(state['features/base/tracks']);
+        const virtualBackground = state['features/virtual-background'];
+
+        try {
+            if (enabled) {
+                await jitsiTrack.setEffect(await getBackgroundEffect(virtualBackground));
+                dispatch(backgroundEnabled());
+            } else {
+                await jitsiTrack.setEffect(undefined);
+                dispatch(blurDisabled());
+            }
+        } catch (error) {
+            dispatch(blurDisabled());
+            logger.error('Error on apply backgroun effect:', error);
+        }
+    };
+}
+
+/**
+ * Signals the local participant that the blur has been enabled.
+ *
+ * @returns {{
+ *      type: BACKGROUND_ENABLED
+ * }}
+ */
+export function backgroundEnabled() {
+    return {
+        type: BACKGROUND_ENABLED
+    };
+}
+
+/**
+ * Signals the local participant that the blur has been disabled.
+ *
+ * @returns {{
+ *      type: BACKGROUND_DISABLED
+ * }}
+ */
+export function blurDisabled() {
```

Comment:
Let's drop this action altogether and pass a parameter to backgroundEnabled instead please.